app/src/main/java/org/oppia/app/home/HomeFragmentController.kt

```diff
@@ -17,20 +22,47 @@ class HomeFragmentController @Inject constructor(
   private val viewModelProvider: ViewModelProvider<UserAppHistoryViewModel>,
   private val userAppHistoryController: UserAppHistoryController
 ) {
+  private var llRoot: LinearLayout? = null
+
   fun handleCreateView(inflater: LayoutInflater, container: ViewGroup?): View? {
     val binding = HomeFragmentBinding.inflate(inflater, container, /* attachToRoot= */ false)
     // NB: Both the view model and lifecycle owner must be set in order to correctly bind LiveData elements to
     // data-bound view models.
     binding.let {
       it.viewModel = getUserAppHistoryViewModel()
-      it.lifecycleOwner = fragment
+      it.lifecycleOwner = fragment as LifecycleOwner?
     }
-
     userAppHistoryController.markUserOpenedApp()
-
+    llRoot = binding.root.findViewById(R.id.llRoot)
+    addContentCard("edit_text_background_border",1)
+    addContentCard("edit_text_background_border_",2)
     return binding.root
   }
 
+  private fun addContentCard(placeholder: String, rows: Int) {
+    val contentComponent = TextInputInteractionView(
+      fragment.context!!,
+      placeholder,
+      rows
+    )
+    val params = LinearLayout.LayoutParams(
+      LinearLayout.LayoutParams.MATCH_PARENT,
+      LinearLayout.LayoutParams.WRAP_CONTENT
+    )
+
+    params.setMargins( dpToPx(8), dpToPx(8),  dpToPx(8), dpToPx(8))
+
+
+    llRoot!!.addView(contentComponent, params)
+
+
+
```

Comment:
Please make sure your PR does not have unnecessary space.

Reply:
resolved